geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,500 @@
+                goodcomps += ["IMAGE {0}".format(output_product)]
+            else:
+                badcomps += ["IMAGE {0}".format(output_product)]
+
+        if self.name == "geotiff_checker":
+            matched_one = True
+            compare_strings += ["GEOTIFF "]
+            if self.module.outputs_match(output_product, compare_product):
+                goodcomps += ["GEOTIFF {0}".format(output_product)]
+            else:
+                badcomps += ["GEOTIFF {0}".format(output_product)]
+
+        if self.name == "text_checker":
+            matched_one = True
+            compare_strings += ["TEXT "]
+            if self.module.outputs_match(self, output_product, compare_product):
+                goodcomps += ["TEXT {0}".format(output_product)]
+            else:
+                badcomps += ["TEXT {0}".format(output_product)]
+
+        if self.name == "netcdf_checker":
+            matched_one = True
+            compare_strings += ["GEOIPS NETCDF "]
+            if self.module.outputs_match(self, output_product, compare_product):
+                goodcomps += ["GEOIPS NETCDF {0}".format(output_product)]
+            else:
+                badcomps += ["GEOIPS NETCDF {0}".format(output_product)]
+
+        if not matched_one:
+            raise TypeError(f"MISSING TEST for output product: {output_product}")
+
+        return goodcomps, badcomps, compare_strings
+
+
+class OutputCheckersInterface(BaseModuleInterface):
+    """Output Checkers routines to apply when comparing data outputs."""
+
+    name = "output_checkers"
+    required_args = {"standard": {}}
+    required_kwargs = {"standard": {}}
+    plugin_class = OutputCheckersBasePlugin
+    # required_args = {
+    #     "standard": ["fname", "output_product", "compare_product"],
+    #     "print_gunzip": ["fobj", "gunzip_fname"],
+    #     "print_gzip": ["fobj", "gzip_fname"],
+    # }
+    # required_kwargs = {"standard": {}}
+    # allowable_kwargs = {
+    #     "test_product": ["goodcomps", "badcomps", "compare_strings"],
+    #     "out_diffname": ["ext", "flag"],
+    #     "compare_outputs": ["test_product_func"],
+    # }
+
+    def identify_checker(self, filename):
+        """Identify the correct output checker plugin and return its name."""
+        checker_found = False
+        checker_name = None
+        for output_checker in self.get_plugins():
+            checker_found = output_checker.module.correct_type(filename)
+            if checker_found:
+                checker_name = output_checker.module.name
+                break
+        if not checker_found:
+            raise TypeError("There isn't an output checker built for this data type.")
+        return checker_name
+
+    def get_plugin(self, name):
+        """Get the output checker plugin corresponding to checker_name and return it."""
+        try:
+            return super().get_plugin(name)
+        except PluginError:
+            return super().get_plugin(self.identify_checker(name))
```

Comment:
So this supports calling get_plugin with either the path to a filename or the name of the plugin?  Do we want to be more explicit here, or is this the "pythonic" way of doing it?

Reply:
@jsolbrig recommended I implement this function in this manner. However, I'm happy to leave more comments in this function so it's not as "black-magic" as it is currently.

Reply:
I'm ok with this, if that's how we want it to work.